build.gradle

```diff
@@ -54,6 +54,27 @@ ext {
     }()
 }
 
+task setVersionAndDateTimeForBuild() {
+    def rootPropertiesFile = new Properties()
+    file("gradle.properties").withInputStream { rootPropertiesFile.load(it) }
+    Date date = new Date()
+    String ricoVersion = rootPropertiesFile.getProperty("version")
+    String buildDate = new java.text.SimpleDateFormat("yyyy-MM-dd").format(date)
+    String buildTime = new java.text.SimpleDateFormat("HH:mm:ss").format(date)
+    String ricoCoreProperties = "base/rico-core/src/main/resources/build.properties"
+    doLast {
+        ant.propertyfile(file: ricoCoreProperties) {
+            entry(key: "version", value: ricoVersion)
+            entry(key: "buildDate", value: buildDate)
+            entry(key: "buildTime", value: buildTime)
+
+        }
+        ant.replaceregexp(file: ricoCoreProperties, match: "^#.*\n", replace: "")
```

Comment:
as many are not fluent in RegExp, please add a comment about what "^#.*\n" matches? If I got it right, it removes all comments, right? :)

Reply:
I'm no expert either but it only matches to the first line of comments. Below you can see a simple RegExp tester window.

As I said I'm doing this because `ant.propertyfile ` automatically adds timestamps as a first line and I'm removing it. 

![image](https://user-images.githubusercontent.com/2084794/95498342-f59da280-09a3-11eb-827e-051a2d726cfd.png)

You'll see the same result over at [regexr.com](https://regexr.com)

I'll include a comment. Thanks for feedback. 

Some tests failed by the way, haven't looked - did it broke something?

Reply:
Just wondering, if we exclude the file via .gitignore. Can we then leave the comment in?

If yes, you could remove the entire regexp

Reply:
Yes, if we do not care whether that timestamp comment line is there or not we don't need regexp. It's just a weird side effect with no option to disable.